test/task_result_buffer_tests.jl

```diff
@@ -128,33 +139,48 @@ end
     @test length(buf.cols[3].elements) == 20
     @test length(buf.row_statuses) == 0
     @test length(buf.row_statuses.elements) == 20
-    @test size(buf.column_indicators) == (0, 3)
+    @test size(buf.missing_values) == (0, 3)
+    @test size(buf.errored_values) == (0, 3)
 end
 
 
 @testset "ColumnIterator" begin
     buf = ChunkedCSV.TaskResultBuffer(1, [Int, Float64], 10)
-    for i in 1:4; push!(buf.cols[1], i); end
-    for i in 1:4; push!(buf.cols[2], Float64(i)); end
+    for i in 1:6; push!(buf.cols[1], i); end
+    for i in 1:6; push!(buf.cols[2], Float64(i)); end
     push!(buf.row_statuses, ChunkedCSV.RowStatus.Ok)
-    push!(buf.row_statuses, ChunkedCSV.RowStatus.HasColumnIndicators)
+    push!(buf.row_statuses, ChunkedCSV.RowStatus.MissingValues)
     push!(buf.row_statuses, ChunkedCSV.RowStatus.TooManyColumns)
-    push!(buf.row_statuses, ChunkedCSV.RowStatus.HasColumnIndicators | ChunkedCSV.RowStatus.ValueParsingError)
-    ChunkedCSV.addrows!(buf.column_indicators, 2)
-    buf.column_indicators[1, 1] = true
-    buf.column_indicators[2, 2] = true
+    push!(buf.row_statuses, ChunkedCSV.RowStatus.MissingValues | ChunkedCSV.RowStatus.ValueParsingError)
+    push!(buf.row_statuses, ChunkedCSV.RowStatus.MissingValues | ChunkedCSV.RowStatus.TooFewColumns)
+    push!(buf.row_statuses, ChunkedCSV.RowStatus.MissingValues | ChunkedCSV.RowStatus.SkippedRow)
+    ChunkedCSV.addrows!(buf.missing_values, 4)
+    ChunkedCSV.addrows!(buf.errored_values, 2)
+
+    buf.missing_values[1, 1] = true
+    buf.missing_values[2, 2] = true
+    buf.missing_values[3, 1] = true
+    buf.missing_values[4, 1] = true
+    buf.missing_values[4, 2] = true
+
+    buf.errored_values[1, 1] = true
+    buf.errored_values[2, 2] = true
 
     iter_data = collect(ChunkedCSV.ColumnIterator{Int}(buf, 1))
-    @test iter_data[1] == ChunkedCSV.ParsedField(1, false, false)
-    @test iter_data[2] == ChunkedCSV.ParsedField(2, false, true)
-    @test iter_data[3] == ChunkedCSV.ParsedField(3, true, false)
-    @test iter_data[4] == ChunkedCSV.ParsedField(4, true, false)
+    @test iter_data[1] == ChunkedCSV.ParsedField(1, false, false, false)
```

Comment:
This test is a bit hard to follow now (so hard to see it is obviously correct). I'm pretty sure it's right/consistent, but it's taken me 10 minutes to check 😅 

Can we add a comment visualizing the data the `TaskResultBuffer` is supposed to represent maybe? like
```
      1, 1.0
missing, 2.0
...
```

And then maybe also add a comment/aligned to help read the `ParsedField` values, like
```suggestion
    #                                ParsedField(val, invalid, error, missing)
    @test iter_data[1] == ChunkedCSV.ParsedField(  1,   false, false, false  )
```